### PR TITLE
Account for handshake authz in `ServerTransportFilterTests`

### LINKED ISSUE #93748: [CI] ServerTransportFilterTests testRemoteAccessInbound failing
Sometimes reproduces on main with e.g. `./gradlew ':x-pack:plugin:security:test' --tests "org.elasticsearch.xpack.security.transport.ServerTransportFilterTests.testRemoteAccessInbound" -Dtests.seed=1143CE3D4C8495BC -Dtests.locale=sl -Dtests.timezone=Asia/Atyrau -Dtests.iters=100`

**Build scan:**
https://gradle-enterprise.elastic.co/s/axzue2h5nm3yc/tests/:x-pack:plugin:security:test/org.elasticsearch.xpack.security.transport.ServerTransportFilterTests/testRemoteAccessInbound

**Reproduction line:**
```
./gradlew ':x-pack:plugin:security:test' --tests "org.elasticsearch.xpack.security.transport.ServerTransportFilterTests.testRemoteAccessInbound" -Dtests.seed=1143CE3D4C8495BC -Dtests.locale=sl -Dtests.timezone=Asia/Atyrau -Druntime.java=17
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.security.transport.ServerTransportFilterTests&tests.test=testRemoteAccessInbound

**Failure excerpt:**
```
org.mockito.exceptions.verification.junit.ArgumentsAreDifferent: 
Argument(s) are different! Wanted:
authorizationService.authorize(
    Authentication[effectiveSubject=Subject{version=8080099, user=User[username=mhQlcGCb,roles=[EXnINfg],fullName=null,email=null,metadata={}], realm={Realm[oidc.tNvavZeK] on Node[vSW]}, type=USER, metadata={}},type=REALM],
    "internal:transport/handshake",
    Mock for TransportRequest, hashCode: 1377995608,
    <any org.elasticsearch.action.ActionListener>
);
-> at org.elasticsearch.xpack.security.transport.ServerTransportFilterTests.testRemoteAccessInbound(ServerTransportFilterTests.java:108)
Actual invocations have different arguments:
authorizationService.authorize(
    Authentication[effectiveSubject=Subject{version=8080099, user=User[username=_system,roles=[],fullName=null,email=null,metadata={}], realm={Realm[__attach.__attach] on Node[]}, type=USER, metadata={}},type=INTERNAL],
    "internal:transport/handshake",
    Mock for TransportRequest, hashCode: 1377995608,
    org.elasticsearch.action.support.PlainActionFuture@262d457a
);
-> at org.elasticsearch.xpack.security.transport.ServerTransportFilter.lambda$inbound$0(ServerTransportFilter.java:110)

  at __randomizedtesting.SeedInfo.seed([1143CE3D4C8495BC:CD51C45ED2A7B404]:0)
  at org.elasticsearch.xpack.security.transport.ServerTransportFilterTests.testRemoteAccessInbound(ServerTransportFilterTests.java:108)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

### PR DESCRIPTION
**Pull request by n1v0lg**
`ServerTransportFilter` uses a different authorization path for the handshake action, by switching to `_system` user context. This is not reflected in `ServerTransportFilterTests` causing failures when the handshake action is selected.

Closes: https://github.com/elastic/elasticsearch/issues/93748

### CODE REVIEW

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterTests.java [hunk 1]
```diff
@@ -328,4 +342,15 @@ private RemoteAccessServerTransportFilter getNodeRemoteAccessFilter(Set<String>
             new SecurityContext(settings, threadContext)
         );
     }
+
+    private static Authentication replaceWithInternalUserAuthcForHandshake(String action, Authentication authentication) {
+        return Set.of(TransportService.HANDSHAKE_ACTION_NAME, TransportActionProxy.getProxyAction(TransportService.HANDSHAKE_ACTION_NAME))
```
Comment (n1v0lg) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterTests.java:347]: Off the top of my head, I don't think the handshake action is ever proxied, so we should tweak the allowlist but I think it makes sense to make this fix separately from addressing the general test failure.

**Comment by n1v0lg** [comment]
Still not quite it, working on fixing another corner case...

**Review by ywangd — APPROVED** [review]
LGTM

**Comment by n1v0lg** [comment]
@elasticmachine update branch plz

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Pull request merged**